app/controllers/openid_application_controller.rb

```diff
@@ -51,6 +56,10 @@ def establish_session
     @current_user = OpenidUser.build_from_identity(identity: user_identity, ttl: ttl)
     @session = build_session(token, token_identifiers.uuid, ttl)
     @session.save && user_identity.save && @current_user.save
+  rescue JWT::ExpiredSignature
+    raise ExpiredTokenError
+  rescue ExpiredTokenError
+    raise ExpiredTokenError
   rescue => e
     Rails.logger.warn(e)
```

Comment:
Can anybody explain to me why we are swallowing errors here? Validation errors get raised and then rescued and then logged, but I have no clue why.

Reply:
When validate_token raises an error, it will jump to the rescue block and skip over establishing the `@session` object and saving it to Redis. Then when it returns to `authenticate_token`, `@session` is nil, so that method returns false, and causes the `authenticate` method to render_unauthorized, which will generate the right HTTP status code.

If you don't want to swallow the error, that's fine, but some caller has to do something meaningful with it, (which could be for example, rendering the unauthorized error with more detail about why, such as "token is expired".

Reply:
If you just removed the `rescue`, I think it would probably result in a 500 error somewhere in the parent `ApplicationController`'s `rescue_from 'Exception' do |exception|` block.